app/src/main/java/de/danoeh/antennapod/fragment/InboxFragment.java

```diff
@@ -77,20 +85,32 @@ public boolean onMenuItemClick(MenuItem item) {
                 showRemoveAllDialog();
             }
             return true;
+        } else if (item.getItemId() == SortMenuBuilder.SORT_DATE_NEW_TO_OLD_ID) {
+            saveSortOrderAndRefresh(DateSortType.DESC);
+            return true;
+        } else if (item.getItemId() == SortMenuBuilder.SORT_DATE_OLD_TO_NEW_ID) {
+            saveSortOrderAndRefresh(DateSortType.ASC);
+            return true;
         }
         return false;
     }
 
+    private void saveSortOrderAndRefresh(DateSortType type) {
+        sortType = type;
+        UserPreferences.setInboxSortOrder(sortType);
+        loadItems();
+    }
+
     @NonNull
     @Override
     protected List<FeedItem> loadData() {
-        return DBReader.getNewItemsList(0, page * EPISODES_PER_PAGE);
+        return DBReader.getNewItemsList(0, page * EPISODES_PER_PAGE, sortType.name());
```

Comment:
Please pass the (SortOrder) enum here, not a string. Strings are hard to check statically.